images/image.go

```diff
@@ -76,12 +117,15 @@ func SynchronousDelete() DeleteOpt {
 type Store interface {
 	Get(ctx context.Context, name string) (Image, error)
 	List(ctx context.Context, filters ...string) ([]Image, error)
-	Create(ctx context.Context, image Image) (Image, error)
+	Create(ctx context.Context, image Image, opts ...CreateOpt) (Image, error)
```

Comment:
Are these changes necessary if the value is contextual? I understand sending it over the API, but this part of the change may be higher impact and not necessary.

Reply:
Using the ctx is opt-in.

This `CreateOpt` change does have breaking impact as it is ellipsized

Reply:
The whole feature is opt-in. Where is this interface change utilized? Seems cleaner to have the initial change use the context and keep the Go API the same. I might not fully be understanding the feature though, will the epoch change here for different images and update?

Reply:
Yes, the epoch is expected to be changed for different images in a single ctx span.

Reply:
Just looking at it as 
`Create(epoch.WithSourceDateEpoch(ctx, t), image)` 
vs 
`Create(ctx, image, images.CreateWithSourceDateEpoch(t))`

and 
`Update(epoch.WithSourceDateEpoch(ctx, t), image)` 
vs
`UpdateWithOpts(ctx, image, images.UpdateWithSourceDateEpoch(t))`

The context riding is cleaner with less backward compatibility headaches. If we want to add opts to these for a 2.0 interface, we could consider adding that there. Even if we had opts, the epoch still feels contextual from an API perspective, if it was truly image data then it would belong in a label.

I am -1 for making this Go API change now, but we could address it in the future especially for 2.0. The other changes and proto API update lgtm though.

Reply:
Remove the Go API change